src/App.js

```diff
@@ -2,32 +2,34 @@ import React, { Component } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
+import { MovieFilter } from './components/MovieFilter';
 
 export class App extends Component {
-  state = {};
+  constructor() {
+    super();
+    this.state = {
+      filteredMovies: moviesFromServer,
+    };
+    this.applyFilteredList = this.applyFilteredList.bind(this);
+  }
+
+  applyFilteredList(filteredList) {
+    this.setState({
+      filteredMovies: filteredList,
+    });
+  }
```

Comment:
Using constructor and 'bind' you making the solution overcomplicated.
Let's make it more simple. 
You should filter the movies according to the 'query' and then transfer filtered movies to the 'MoviesList' component. The state should be on the top level to give us the possibility to filter films by this query. 
That means that if you want to make a separate component for the 'search input', you need somehow to transfer input value from the child component('MovieFilter') to the parent component, where you will handle that input value for filtering movies. 
Understanding the topic 'lifting state up' will help you a lot.
Long story short: your 'Filter' component should receive a 'query' and 'queryChangeHandler' from the parent. Then this component should handle input changes and transfer them to the top level using lifting state up. There you will be filtering your movies according to the new query.
If you have some questions - write to your channel or visit the q&a. I will explain

Reply:
@varseniuk  Okay, I`ll return to this task after learning about lifting state up.